content/Google is a modal dialog - Martin Rubi.md

```diff
@@ -0,0 +1,116 @@
+in the user interface.
+
+The first desktop apps achieved 3D emulation by overlapping 2D rectangles called
+windows [1].
+
+Browser interfaces avoid 3D emulation and usually stick to a strict 2D interface
+through laying out its elements instead of using overlapping.
+
+However this is just the implementation of what modal dialogs do and not its functional
+purpose and value.
+
+Modal dialogs interrupt the flow of the user attention to solve a particular task
+providing a self contained, highly focused functional and visual context for the user.
+
+That is they provide for a very short period of time a user interface for the user
+to solve a specific task hidding every other complexity not related to that task.
+
+Once the user ends solving that specific part of the app it returns to the previous
+flow just like she left it.
+
+Whichs is pretty much the same workflow that we use to google things
+
+### Google is a modal dialog
+
+How do we google things?
+
+Usually we are doing something like writing a blog post and we find the need to
+search for information.
+
+We open a new tab with a clean and very specific design to do just one thing,
+search things.
+
+We select a few results, decide whether our question was answered or not and get
+back to our previous task closing the search tab.
+
+That is what modal dialogs functional purpose is regardless of its implementation
+on the user interface space.
+
+### Other issues with edit in place
+
+There are two very important functions in a modal dialog.
+
+The first one would be the one already mentioned, to provide an isolated context
+to solve a specific task hidding all other context not relevant to that task.
+
+Yet there is another function just as important: hidding unnecessary complexity
+in the main application interface until that complexity is required.
+
+The rationale of a modal dialog is to present to the user the complexity of
+a user interface when the user needs it, not before, not afterwards, and just for
+the time the user is dealing with that task.
+
+The proliferation of menu options and visual aids can hurt more than they would help
+producing a sense of overwhelming and confusion in the user.
+
+One example of that value are context menus where instead of always showing all options
+to the user they show just the options for a particular topic and only when the user
+requests it, for example after a right button click.
+
+Popup context menus are modal dialogs. So are tooltips.
+
+Edit in place has other problems that modal dialogs solve in a clear and explicit
+way like commiting or rolling back an edition.
+
+A modal dialog requires an explicit action to accept an edition and provides a clear
+interface to dismiss the edition in case the user wants to roll it back.
+
+Edit in place has ctl-z but that produces very different experiencies in the user: when is
+a change applied in a edit in place field, when it loses focus, on every char typed,
```

Comment:
I would change the comma after field for a closing question mark, that is the question in essence, the rest of the question are possible answers. It makes it more readable.